cmd/neutron_query_relayer/main.go

```diff
@@ -61,15 +62,49 @@ func main() {
 	ctx, cancel := context.WithCancel(context.Background())
 	wg := &sync.WaitGroup{}
 
+	// The storage has to be shared because of the LevelDB single process restriction.
+	storage, err := app.NewDefaultStorage(cfg, logger)
+	if err != nil {
+		logger.Fatal("Failed to create NewDefaultStorage", zap.Error(err))
+	}
+	defer func(storage relay.Storage) {
+		if cfg.AllowTxQueries {
```

Comment:
now I reailse that this suggestion was a piece of shit https://github.com/neutron-org/neutron-query-relayer/pull/44#discussion_r1005427108: storage is also used by KV processor, so we are not able to have a nil storage on `AllowTxQueries==false` and therefore should close storage regardless of tx query allowance

Reply:
So, does that mean we **always** need storage to be correctly initialised in any workflow? Why wouldn't we just always initialise it then? That will allow us to remove dummy storage implementation completely, and also remove some useless logic around it.

Reply:
exactly, that's the idea

Reply:
okay, i'll revert that

Reply:
Dummy storage implementation is still present though, don't we need to remove it since we don't use it?